src/plugins/data/public/field_formats/field_formats_service.ts

```diff
@@ -18,62 +18,36 @@
  */
 
 import { CoreSetup } from 'src/core/public';
-import { FieldFormatRegisty } from './field_formats';
-
-import {
-  BoolFormat,
-  BytesFormat,
-  ColorFormat,
-  DateFormat,
-  DateNanosFormat,
-  DurationFormat,
-  IpFormat,
-  NumberFormat,
-  PercentFormat,
-  RelativeDateFormat,
-  SourceFormat,
-  StaticLookupFormat,
-  StringFormat,
-  TruncateFormat,
-  UrlFormat,
-} from '../../common/';
+import { FieldFormatRegisty } from '../../common/field_formats';
 
 export class FieldFormatsService {
   private readonly fieldFormats: FieldFormatRegisty = new FieldFormatRegisty();
 
   public setup(core: CoreSetup) {
-    this.fieldFormats.init(core);
-
-    this.fieldFormats.register([
-      BoolFormat,
-      BytesFormat,
-      ColorFormat,
-      DateFormat,
-      DateNanosFormat,
-      DurationFormat,
-      IpFormat,
-      NumberFormat,
-      PercentFormat,
-      RelativeDateFormat,
-      SourceFormat,
-      StaticLookupFormat,
-      StringFormat,
-      TruncateFormat,
-      UrlFormat,
-    ]);
+    core.uiSettings.getUpdate$().subscribe(({ key, newValue }) => {
+      if (key === 'format:defaultTypeMap') {
+        this.fieldFormats.parseDefaultTypeMap(newValue);
+      }
+    });
+
+    this.fieldFormats.init(core.uiSettings.get, {
+      parsedUrl: {
+        origin: window.location.origin,
+        pathname: window.location.pathname,
+        basePath: core.http.basePath.get(),
+      },
+    });
 
     return this.fieldFormats as FieldFormatsSetup;
   }
 
   public start() {
     return this.fieldFormats as FieldFormatsStart;
   }
-
-  public stop() {
-    // nothing to do here yet
-  }
 }
 
 /** @public */
 export type FieldFormatsSetup = Omit<FieldFormatRegisty, 'init'>;
```

Comment:
I think the contracts returned from public \ server can be the same (and defined in common?) - 
( I checked and no one uses npSetup.register)

```
/** @public */
export type FieldFormatsSetup = Pick<FieldFormatRegisty, 'register'>;

}** @public */
export type FieldFormatsStart = Omit<FieldFormatRegisty, 'init' & 'register'>;
```